nx/source/services/nfc.c

```diff
@@ -0,0 +1,826 @@
+
+    Result rc = serviceIpcDispatch(&g_nfpuInterface);
+
+    if (R_SUCCEEDED(rc)) {
+        IpcParsedCommand r;
+        struct {
+            u64 magic;
+            u64 result;
+        } *resp;
+
+        serviceIpcParse(&g_nfpuInterface, &r, sizeof(*resp));
+        resp = r.Raw;
+
+        rc = resp->result;          
+    }
+
+    return rc;
+}
+
+Result nfpuUnmount(HidControllerID id) {
+    if (id == CONTROLLER_P1_AUTO)
+        return _nfpuInterfaceCmdInIdNoOut(&g_nfpuInterface, 6, g_controllerP1AutoID);
+    return _nfpuInterfaceCmdInIdNoOut(&g_nfpuInterface, 6, id);
+}
+
+Result nfpuGetTagInfo(HidControllerID id, NfpuTagInfo *out) {
+    if (id == CONTROLLER_P1_AUTO)
+        return _nfpuInterfaceCmdInIdOutBuffer(&g_nfpuInterface, 13, g_controllerP1AutoID, out, sizeof(NfpuTagInfo));
+    return _nfpuInterfaceCmdInIdOutBuffer(&g_nfpuInterface, 13, id, out, sizeof(NfpuTagInfo));
+}
+
+Result nfpuGetRegisterInfo(HidControllerID id, NfpuRegisterInfo *out) {
+    if (id == CONTROLLER_P1_AUTO)
+        return _nfpuInterfaceCmdInIdOutBuffer(&g_nfpuInterface, 14, g_controllerP1AutoID, out, sizeof(NfpuRegisterInfo));
+    return _nfpuInterfaceCmdInIdOutBuffer(&g_nfpuInterface, 14, id, out, sizeof(NfpuRegisterInfo));
+}
+
+Result nfpuGetCommonInfo(HidControllerID id, NfpuCommonInfo *out) {
+    if (id == CONTROLLER_P1_AUTO)
+        return _nfpuInterfaceCmdInIdOutBuffer(&g_nfpuInterface, 15, g_controllerP1AutoID, out, sizeof(NfpuCommonInfo));
+    return _nfpuInterfaceCmdInIdOutBuffer(&g_nfpuInterface, 15, id, out, sizeof(NfpuCommonInfo));
+}
+
+Result nfpuGetModelInfo(HidControllerID id, NfpuModelInfo *out) {
+    if (id == CONTROLLER_P1_AUTO)
+        return _nfpuInterfaceCmdInIdOutBuffer(&g_nfpuInterface, 16, g_controllerP1AutoID, out, sizeof(NfpuModelInfo));
+    return _nfpuInterfaceCmdInIdOutBuffer(&g_nfpuInterface, 16, id, out, sizeof(NfpuModelInfo));
+}
+
+Result nfpuOpenApplicationArea(HidControllerID id, u32 app_id, u32 *npad_id) {
+    if (id == CONTROLLER_P1_AUTO)
+        return nfpuOpenApplicationArea(g_controllerP1AutoID, app_id, npad_id);
+
+    IpcCommand c;
+    ipcInitialize(&c);
+
+    struct {
+        u64 magic;
+        u64 cmd_id;
+        u64 id;
+        u32 app_id;
+    } *raw;
+
+    raw = serviceIpcPrepareHeader(&g_nfpuInterface, &c, sizeof(*raw));
+
+    raw->magic = SFCI_MAGIC;
+    raw->cmd_id = 7;
+    raw->id = hidControllerIDToOfficial(id);
+    raw->app_id = app_id;
+
+    Result rc = serviceIpcDispatch(&g_nfpuInterface);
+
+    if (R_SUCCEEDED(rc)) {
+        IpcParsedCommand r;
+        struct {
+            u64 magic;
+            u64 result;
+            u32 npad_id;
+        } *resp;
+
+        serviceIpcParse(&g_nfpuInterface, &r, sizeof(*resp));
+        resp = r.Raw;
+
+        rc = resp->result; 
+
+        if (R_SUCCEEDED(rc) && npad_id)
+            *npad_id = resp->npad_id;         
+    }
+
+    return rc;
+}
+
+Result nfpuGetApplicationArea(HidControllerID id, void *buf, size_t buf_size) {
```

Comment:
Update with the same change as the .h, same below.

Reply:
Should all the unaries match this?

Reply:
Just void and Service.